### PR TITLE
[Admin] Adds replays to all past notes and future ones, Adds view replay button for clients

### PR DESCRIPTION
**Pull request by JamieD1**
Adds replays to all past notes and future ones, Adds view replay button for clients

#### Changelog

:cl:  
rscadd: Adds replays to all past notes and future ones, Adds view replay button for clients
/:cl:

### CODE REVIEW

**Comment by Hopekz** [comment]
Good PR

File: interface/interface.dm [hunk 1]
```diff
@@ -27,6 +27,21 @@
 		to_chat(src, "<span class='danger'>The forum URL is not set in the server configuration.</span>")
 	return
 
+/client/verb/demoview(roundnumber as num)
+	set name = "demoview"
+	set desc = "Enter Round Number you wish to see the replay of"
+	set hidden = 1
```
Comment (Bibby0110) [inline interface/interface.dm:33]:
```suggestion
	set hidden = TRUE
```

File: interface/interface.dm [hunk 2]
```diff
@@ -27,6 +27,21 @@
...
+	var/roundurl = CONFIG_GET(string/demourl)
+	if(roundurl)
+		if(roundnumber)
+			if(alert("This will open the demo in your browser. Are you sure?",,"Yes","No")!="Yes")
```
Comment (Bibby0110) [inline interface/interface.dm]: Move above the if(roundnumber) 
Currently it only asks if you've entered a round ID, if you haven't you just get sent without confirmation

Maybe add an isnum check for both GLOB.round_id - 1 and var/roundnumber incase DB is down or other weird fuckery

File: interface/skin.dmf [hunk 3]
```diff
@@ -116,7 +116,6 @@ window "mainwindow"
 		anchor2 = none
 		is-visible = false
 		saved-params = ""
-		auto-format = false
```
Comment (Bibby0110) [inline interface/skin.dmf]: Don't remove that

File: yogstation/code/modules/admin/sql_message_system.dm [hunk 4]
```diff
@@ -370,8 +371,8 @@
 						alpha = 10
 						skipped = TRUE
 					alphatext = "filter: alpha(opacity=[alpha]); opacity: [alpha/100];"
-
-			var/list/data = list("<p style='margin:0px;[alphatext]'> <b>[timestamp] | [server] | [admin_key]")
+			var/list/data = list("<p style='margin:0px;[alphatext]'> <b>[timestamp] | [server] | [admin_key] | ")
+			data += "Round: #<a href='?_src_=holder;[HrefToken()];viewdemo=[roundnumber]'>[roundnumber]</a>"
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/sql_message_system.dm:375]: Extra thing, find the first stored replays manually and check that the round number >= to that so people don't attempt to view replays from pre-replay days

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -811,6 +811,10 @@
 			var/message_id = href_list["deletemessageempty"]
 			delete_message(message_id, browse = TRUE)
 
+	else if(href_list["viewdemo"])
+		var/roundnumber = href_list["viewdemo"]
```
Comment (Bibby0110) [inline code/modules/admin/topic.dm:815]: verify the input, isnum()

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

**Pull request merged**